commit 985e88b13a25599eeb46b03c123531a5a11ac1cd
Author: Bob Copeland <me@bobcopeland.com>
Date:   Thu Jan 22 16:49:53 2015 -0500

    Revert "mac80211: keep sending peer candidate events while in listen state"
    
    This reverts commit 2ae70efcea7a695a62bb47170d3fb16674b8dbea.
    
    The new peer events that are generated by the change are causing problems
    with wpa_supplicant in userspace: wpa_s tries to restart SAE authentication
    with the peer when receiving the event, even though authentication may be in
    progress already, and it gets very confused.
    
    Revert back to the original operating mode, which is to only get events when
    there is no corresponding station entry.
    
    Cc: Nishikawa, Kenzoh <Kenzoh.Nishikawa@jp.sony.com>
    Cc: Masashi Honma <masashi.honma@gmail.com>
    Signed-off-by: Bob Copeland <me@bobcopeland.com>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/mesh_plink.c b/net/mac80211/mesh_plink.c
index fa94ca15ba95..b488e1859b18 100644
--- a/net/mac80211/mesh_plink.c
+++ b/net/mac80211/mesh_plink.c
@@ -523,13 +523,6 @@ void mesh_neighbour_update(struct ieee80211_sub_if_data *sdata,
 	    sdata->u.mesh.mshcfg.auto_open_plinks &&
 	    rssi_threshold_check(sdata, sta))
 		changed = mesh_plink_open(sta);
-	else if (sta->plink_state == NL80211_PLINK_LISTEN &&
-		 (sdata->u.mesh.user_mpm ||
-		  sdata->u.mesh.security & IEEE80211_MESH_SEC_AUTHED))
-		cfg80211_notify_new_peer_candidate(sdata->dev, hw_addr,
-						   elems->ie_start,
-						   elems->total_len,
-						   GFP_ATOMIC);
 
 	ieee80211_mps_frame_release(sta, elems);
 out: